src/eradiate/cli.py

```diff
@@ -292,5 +305,204 @@ def info(data_stores, list_registry):
+)
+@click.option(
+    "-d", "--dry-run", is_flag=True, help="Do not write filtered data to disk."
+)
+@click.option("-v", "--verbose", is_flag=True, help="Display filtering summary.")
+def trim(
+    filename: str,
+    output: str,
+    interactive: bool = False,
+    dry_run: bool = False,
+    verbose: bool = False,
+):
+    """
+    Trim a spectral response function.
+
+    Remove all-except-last leading zeros and all-except-first trailing zeros.
+
+    If the option --destination is not specified, the pruned data is written in
+    the same directory as PATH with '-pruned' appended to the original filename.
+
+    FILENAME is the path to the spectral response function data set to process.
+    OUTPUT specifies where to write the filtered data set.
+    """
+    trimmed = trim_srf(srf=filename, verbose=False)
+
+    if verbose:
+        table = summarize(srf=filename, filtered=trimmed)
+        console.print("\n", table, "\n")
+
+    if interactive:
+        show_srf(
+            ds=filename,
+            title=filename,
+            trim_ds=False,
+            threshold=None,
+            wmin=f"{to_quantity(trimmed.w).min()}:~",
+            wmax=f"{to_quantity(trimmed.w).max()}:~",
+            percentage=None,
+        )
+        if click.confirm("Are you sure you want to apply this filter?", abort=True):
+            write_filtered_data(path=output, filtered=trimmed, dry_run=dry_run)
+
+    else:
+        write_filtered_data(path=output, filtered=trimmed, dry_run=dry_run)
+
+
+@srf.command()
+@click.argument("filename", type=click.Path(exists=True))
+@click.argument("output", type=click.Path())
+@click.option(
+    "--trim/--no-trim",
+    default=True,
+    show_default=True,
+    help="Trim the data set prior to filtering.",
+)
+@click.option(
+    "-v",
+    "--verbose",
+    is_flag=True,
+    help="Display filtering summary.",
+)
+@click.option(
+    "-d",
+    "--dry-run",
+    is_flag=True,
+    help="Do not write filtered data to file.",
+)
+@click.option(
+    "-i",
+    "--interactive",
+    is_flag=True,
+    help="Prompt before writing filtered data to disk.",
+)
+@click.option(
+    "-t",
+    "--threshold",
+    type=float,
+    help="Data points where response is less then or equal to this value are dropped.",
+)
+@click.option(
+    "-w",
+    "--wmin",
+    help="Lower wavelength value",
+)
+@click.option(
+    "-W",
+    "--wmax",
+    help="Upper wavelength value",
+)
+@click.option(
+    "-p",
+    "--percentage",
+    type=float,
+    help="Data points that do not contribute to this percentage of the integrated spectral response are dropped",
+)
+def filter(
+    filename: str,
+    output: str,
+    trim: bool = True,
+    verbose: bool = False,
+    dry_run: bool = False,
+    interactive: bool = False,
+    threshold: t.Optional[float] = None,
+    wmin: t.Optional[t.Union[float, str]] = None,
+    wmax: t.Optional[t.Union[float, str]] = None,
+    percentage: t.Optional[float] = None,
```

Comment:
Type hints, as well as default values, are unnecessary and even confusing IMO: parameter values are entirely controlled by the command-line handling, so all this information is anyway superseded by what you'll pass to the option and argument decorators. For consistency with the rest of the CLI code, I'd prefer if you'd remove those.

Note that if, as discussed live, you want to be able to call these functions outside the CLI (*e.g.* in a notebook), then the correct way to do it in our codebase is to transfer the implementation to a relevant module (*e.g.* `eradiate.srf_tools`) and make this function a simple wrapper.